_playwright-tests/style_guide.md

```diff
@@ -0,0 +1,238 @@
+}, {
+  message: 'Expected API to return 200',
+  timeout: 10000,
+}).toBe(200);
+```
+
+For assertion of one out of many or multiple checks at once, you can also use the `race` and `all` from the JS promises.
+Example:
+```typescript
+await Promise.all([
+  page.waitForResponse(
+    (resp) =>
+      resp.url().includes('bulk_delete') && resp.status() >= 200 && resp.status() < 300,
+    ),
+  page.getByRole('button', { name: 'Remove' }).click(),
+]);
+```
+
+## Debugging 🐛
+
+Playwright offers several powerful tools for debugging your tests (ex.: headed/debug mode, trace/ viewer, codegen). These tools can be accessed through Playwright's CLI or IDE extension. Some examples:
+
+### Playwright Inspector
+The Inspector is a GUI tool that helps you step through your tests, see what Playwright is doing, and explore selectors.
+The editable selector explorer at the bottom of the page can be really usefull for crafting good selectors. 💡
+
+Run your tests with the `--debug` flag to open the Inspector:
+```bash
+npx playwright test --debug
+```
+
+### Trace Viewer
+The Trace Viewer provides a detailed trace of your test execution, including screenshots, actions, and logs. It's invaluable for diagnosing failures after a test run.
+
+Enable it in your `playwright.config.ts`:
+```typescript
+import { defineConfig } from '@playwright/test';
+
+export default defineConfig({
+  use: {
+    trace: 'on', // or 'on-first-retry', 'retain-on-failure'
+  },
+});
+```
+Then view the trace with:
+```bash
+npx playwright show-trace trace.zip
+```
+
+### `page.pause()`
+Insert `await page.pause()` into your test to pause execution and open a browser window with debugging tools. This allows you to inspect the page state at a specific point in your test.
+
+### VS Code Extension
+The official [Playwright Test for VS Code](https://playwright.dev/docs/getting-started-vs-code) extension offers a great debugging experience directly within your editor, including running tests with a single click, setting breakpoints, and live debugging.
+But don't worry, the vast majority if not everything can be achieved through the CLI, so you can use your preffered editor of choice. 
+
+## Caveats, gotchas and things to be aware of ⚠
+
+- When working with PF modals/wizards/dialogs, there is a problem where they can incorrectly 'hide'/disable the page behind them.
+
+  - This can cause issues with Playwright, as it might try to interact with elements that are/should not visible or enabled.
+  - To work around this, you can target the dialog and save it to a variable, then use that variable to interact with any elements inside the dialog.
+
+    ```typescript
+    const modal = page.getByRole('dialog', { name: 'Title' });
+    modal.getByRole('button', { name: 'Submit' }).click();
+    ```
+
+- Navigation inside Insights (stage and proxy especially) can be flaky.
+  - This can be caused by multiple reasons (stage/proxy instability, sentry errors, cache failures, slooow loads 🐌).
```

Comment:
```suggestion
  - This can be caused by multiple reasons (stage or proxy instability, sentry errors, cache failures, slooow loads 🐌).
```
Try hard to avoid the use of slash in text. Take a moment to figure out if you mean "or" or "and".